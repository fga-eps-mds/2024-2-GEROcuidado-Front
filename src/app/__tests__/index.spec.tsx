import React from "react";
import { render } from "@testing-library/react-native";
import Home from "..";

describe("Home", () => {
  it("renderiza corretamente", () => {
    const { toJSON } = render(<Home />);
    expect(toJSON()).toMatchSnapshot();
  });
  it("deve conter um título", () => {
    const { getByText } = render(<Home />);
    const titleElement = getByText("Seja um GEROcuidador!");
    expect(titleElement).toBeTruthy();
  });

  it("deve conter botões de ação", () => {
    const { getByText } = render(<Home />);
    const forumButton = getByText("Acessar Fórum");
    const loginButton = getByText("Login");
    const cadastroButton = getByText("Cadastre-se");
    expect(forumButton).toBeTruthy();
    expect(loginButton).toBeTruthy();
    expect(cadastroButton).toBeTruthy();
  });
});
