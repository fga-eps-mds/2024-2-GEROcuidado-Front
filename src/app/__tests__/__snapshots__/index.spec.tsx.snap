// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renderiza corretamente 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
  />
  <Text
    style={
      {
        "fontSize": 37,
        "fontWeight": "700",
        "margin": 20,
        "marginBottom": 70,
        "textAlign": "center",
      }
    }
  >
     Seja um GEROcuidador! 
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2CCDB5",
        "borderRadius": 20,
        "marginBottom": 25,
        "maxWidth": 350,
        "paddingHorizontal": 26,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
    testID="link-button"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "700",
        }
      }
    >
      Acessar FÃ³rum
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2CCDB5",
        "borderRadius": 20,
        "marginBottom": 25,
        "maxWidth": 350,
        "paddingHorizontal": 26,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
    testID="link-button"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "700",
        }
      }
    >
      Login
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#B47B9D",
        "borderRadius": 20,
        "marginBottom": 25,
        "maxWidth": 350,
        "paddingHorizontal": 26,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
    testID="link-button"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "700",
        }
      }
    >
      Cadastre-se
    </Text>
  </View>
</View>
`;
